pkg/cmd/cli/restic/server.go

```diff
@@ -191,3 +202,50 @@ func (s *resticServer) run() {
 	s.logger.Info("Waiting for all controllers to shut down gracefully")
 	wg.Wait()
 }
+
+// validatePodVolumesHostPath validates that the pod volumes path contains a
+// directory for each Pod running on this node
+func (s *resticServer) validatePodVolumesHostPath() error {
+	files, err := s.fileSystem.ReadDir("/host_pods/")
+	if err != nil {
+		return errors.Wrap(err, "could not read pod volumes host path")
+	}
+
+	// create a map of directory names inside the pod volumes path
+	dirs := sets.NewString()
+	for _, f := range files {
+		if f.IsDir() {
+			dirs.Insert(f.Name())
+		}
+	}
+
+	pods, err := s.kubeClient.CoreV1().Pods("").List(metav1.ListOptions{FieldSelector: fmt.Sprintf("spec.nodeName=%s,status.phase=Running", os.Getenv("NODE_NAME"))})
```

Comment:
unfortunately there isn't a great way to unit test the field selector (https://github.com/kubernetes/client-go/issues/326). Seems like it might be possible to mimic the functionality by creating a custom reactor, but then that won't really test that this field selector is correct.